app/controllers/spree/admin/payments_controller_decorator.rb

```diff
@@ -5,7 +5,8 @@
   # When a user fires an event, take them back to where they came from
   # (we can't use respond_override because Spree no longer uses respond_with)
   def fire
-    return unless event = params[:e] && @payment.payment_source
+    # Unfortunately, due to the assignment, this needs to be an 'and' rather than an '&&'
+    return unless event = params[:e] and @payment.payment_source # rubocop:disable Style/AndOr
```

Comment:
We should add a cop to prevent conditional assignment IMHO. Can we remove it splitting this line? It's both difficult to read and dangerously brittle.